src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalVariableNameCheck.java

```diff
@@ -116,7 +117,7 @@ protected final boolean mustCheckName(DetailAST ast) {
             final boolean isFinal = modifiersAST.findFirstToken(TokenTypes.FINAL) != null;
             result = !isFinal && ScopeUtil.isLocalVariableDef(ast);
         }
-        return result;
+        return result && !TokenUtil.isUnnamed(ast.findFirstToken(TokenTypes.IDENT));
```

Comment:
There are 2 ways to handle this issue. Either we update the existing property regexp to allow underscores, or we update the code to fully ignore single underscore variable names.

The regexp allows anyone to change it and what it accepts. Changing it would allow the users to backward compatible this change.

Change the code hard codes that this is to be a full on ignore which can't be undone. This is similar to ignore `final` variables, as another check handles this. If someone wanted to configure CS to violate underscore variables again, they would have to use another check, `IllegalTokenText` or possibly `IllegalIdentifierName`.

Are we sure we want to basically break compatibility and hard code the ignore? I was originally thinking we would just update the property. I am leaning more to the property just to remain compatible with those not on 21 yet. I remember somewhere we said we wanted to keep backwards compatibility with another check.

@nrmancuso Feel free to review.

Reply:
> I was originally thinking we would just update the property.

Yes, please do this @mahfouz72